src/platform/gpu.h

```diff
@@ -1,11 +1,117 @@
 #pragma once
 
+#include "common/aliases.h"
 #include "platform/native_platform.h"
 
 namespace vx
 {
+struct gpu_buffer;
+struct gpu_texture;
+struct gpu_sampler;
+struct gpu_vertex_desc;
+struct gpu_shader;
+struct gpu_pipeline;
 struct gpu_channel;
 
+enum gpu_vertex_format
```

Comment:
what do you think about `enum class`? It scopes the enum value. It doesn't change the code very much, since we'd be doing `gpu_vertex_format::float2` instead of `gpu_vertex_format_float2`, but it does have the benefit of not leaking the enum values into the `vx` namespace 🙂